Automattic/Gravatar-SDK-Android
# Adam/oauth user email

### Description

This PR updates the OAuth flow: 
1. `user_email` param is added to the URL to pre-fill the login form.
2. The `OAuthPage` UI is updated to match the design of the QE.

<img width="334" alt="Screenshot 2024-09-03 at 13 41 49" src="https://github.com/user-attachments/assets/092e0245-3ff1-4dba-b4df-dcf1cb386f0f">

The copy is from me and it's based on the invalid session error. I'm still not sure whether we should say to the user to "log in" or "authorize $app_name" kind of message. What do you think?

### Testing Steps

1. Start the demo app
2. Logout the user
3. Logout from the WP in the browser 
4. Start the QE
5. Confirm you see an email pre-filled
6. Tap "x" in the top bar
7. Confirm you see the update UI
8. Tap the "Log in" 
9. Finish the OAuth flow
10. Confirm it worked
